.github/workflows/deploy-sharing-smart-contract.yml

```
@@ -0,0 +1,161 @@
+        working-directory: packages/sharing-smart-contract
+        env:
+          WALLET_PRIVATE_KEY: ${{ secrets.DEPLOYER_DEV_PRIVATEKEY }}
+          ENV: staging
+          MANIFEST_DEFAULT_DIR: .openzeppelin/staging
+        run: |
+          npm run deploy -- --network bellecour
+          git add .openzeppelin
+
+      - name: Deploy Contracts (Prod)
+        if: ${{ github.event.inputs.target == 'sharing-smart-contract-deploy-prod' && github.ref == 'refs/heads/main' }}
+        working-directory: packages/sharing-smart-contract
+        env:
+          WALLET_PRIVATE_KEY: ${{ secrets.DEPLOYER_PROD_PRIVATEKEY }}
+          MANIFEST_DEFAULT_DIR: .openzeppelin/prod
+        run: |
+          npm run deploy -- --network bellecour
+          git add .openzeppelin
+
+      - name: Update Production Environment
+        if: ${{ github.event.inputs.target == 'sharing-smart-contract-deploy-prod' }}
+        working-directory: environments
+        run: |
+          KEY=dataprotectorSharingContractAddress VALUE=$(cat ../packages/sharing-smart-contract/deployments/DataProtectorSharing/address) npm run update-env
+          KEY=dataprotectorSharingStartBlock VALUE=$(cat ../packages/sharing-smart-contract/deployments/DataProtectorSharing/block) npm run update-env
+          KEY=addOnlyAppWhitelistRegistryContractAddress VALUE=$(cat ../packages/sharing-smart-contract/deployments/AddOnlyAppWhitelistRegistry/address) npm run update-env
+          KEY=addOnlyAppWhitelistRegistryStartBlock VALUE=$(cat ../packages/sharing-smart-contract/deployments/AddOnlyAppWhitelistRegistry/block) npm run update-env
+          git add environments.json
+          git commit -m "Deployment ${{ github.event.inputs.target }} run ${{ github.run_number }} commit ${{ github.sha }}" --author="drone-product <team-product@iex.ec>"
+
+      - name: Update Staging Environment
+        if: ${{ github.event.inputs.target == 'sharing-smart-contract-deploy-staging' }}
+        working-directory: environments
+        run: |
+          KEY=dataprotectorSharingContractAddress VALUE=$(cat ../packages/sharing-smart-contract/deployments/DataProtectorSharing/address) npm run update-env
+          KEY=dataprotectorSharingStartBlock VALUE=$(cat ../packages/sharing-smart-contract/deployments/DataProtectorSharing/block) npm run update-env
+          KEY=addOnlyAppWhitelistRegistryContractAddress VALUE=$(cat ../packages/sharing-smart-contract/deployments/AddOnlyAppWhitelistRegistry/address) npm run update-env
+          KEY=addOnlyAppWhitelistRegistryStartBlock VALUE=$(cat ../packages/sharing-smart-contract/deployments/AddOnlyAppWhitelistRegistry/block) npm run update-env
+          git add environments.json
+          git commit -m "Deployment ${{ github.event.inputs.target }} run ${{ github.run_number }} commit ${{ github.sha }}" --author="drone-product <team-product@iex.ec>"
```

```suggestion
          git commit -m "Deployment ${{ github.event.inputs.target }} run ${{ github.run_number }} commit ${{ github.sha }}" --author="iExec Github Action <ci@iex.ec>"
```
Or something like that.